### PR TITLE
docs: various small changes

### PR DESCRIPTION
**Pull request by zuercher**
Commit Message:
* Note need to install a sufficiently recent libstdc++ version when
setting up clang builds. For example, Ubuntu gcc packages install
libstdc++, but depending on the gcc version libstdc++ might not be
new enough to compile Envoy.
* Fix links in HTTP conn manager protos.
* Update RELEASES.md with previous release.

Additional Description:
re: libstd++: https://envoyproxy.slack.com/archives/C7E6C71QB/p1586307917037300

Risk Level: low, docs only
Testing: n/a
Docs Changes: yes
Release Notes: n/a

Signed-off-by: Stephan Zuercher <zuercher@gmail.com>

### CODE REVIEW

**Comment by zuercher** [comment]
I found a few other small things and decided to roll them into this PR instead of spinning several.

File: bazel/README.md [hunk 1]
```diff
@@ -80,6 +80,9 @@ for how to update or override dependencies.
     echo "build --config=clang" >> user.bazelrc
     ```
 
+    Note: `libstdc++-7-dev` (or higher) must be installed. This is typically available via a
```
Comment (lizan) [inline bazel/README.md:83]: This isn't true if you're using libc++, the alternative is use `--config=libc++`.

Comment (zuercher) [reply bazel/README.md:83]: So should it say install libstdc++-7-dev or use `--config=libc++`?

Comment (lizan) [reply bazel/README.md:83]: Yes

**Comment by kyessenov** [comment]
This PR wiped out go-control-plane stubs somehow https://github.com/envoyproxy/go-control-plane/commit/2f1cfbd16dfb8cc975a895006b7bbfcd386e8536. Was there something that changed in CI at the same time?

**Pull request merged**